test.tf

```diff
@@ -108,24 +123,99 @@ resource "azurerm_postgresql_server" "example" {
   name                = "connection_throttling"
   resource_group_name = "testrg"
   server_name         = azurerm_postgresql_server.example.name
-  value               = "off"
+  value               = "on"
 }
 
 resource "azurerm_postgresql_configuration" "example" {
   name                = "log_checkpoints"
   resource_group_name = "testrg"
   server_name         = azurerm_postgresql_server.example.name
-  value               = "off"
+  value               = "on"
+}
+data "azurerm_client_config" "example_clientconfig" {}
+resource "azurerm_postgresql_active_directory_administrator" "example_adconfig" {
+  server_name         = azurerm_postgresql_server.example.name
+  resource_group_name = azurerm_resource_group.resource_group.name
+  login               = "sqladmin"
+  tenant_id           = data.azurerm_client_config.example_clientconfig.tenant_id
+  object_id           = data.azurerm_client_config.example_clientconfig.object_id
+}
+data "azurerm_client_config" "example_clientconfig" {}
+resource "azurerm_postgresql_active_directory_administrator" "example_adconfig" {
+  server_name         = azurerm_postgresql_server.example.name
+  resource_group_name = azurerm_resource_group.resource_group.name
+  login               = "sqladmin"
+  tenant_id           = data.azurerm_client_config.example_clientconfig.tenant_id
+  object_id           = data.azurerm_client_config.example_clientconfig.object_id
+}
+data "azurerm_client_config" "example_clientconfig" {}
+resource "azurerm_mysql_active_directory_administrator" "example_adconfig" {
+  server_name         = azurerm_mysql_server.example.name
+  resource_group_name = azurerm_resource_group.resource_group.name
+  login               = "sqladmin"
+  tenant_id           = data.azurerm_client_config.example_clientconfig.tenant_id
+  object_id           = data.azurerm_client_config.example_clientconfig.object_id
+}
+data "azurerm_client_config" "example_clientconfig" {}
+resource "azurerm_mysql_active_directory_administrator" "example_adconfig" {
+  server_name         = azurerm_mysql_server.example.name
+  resource_group_name = azurerm_resource_group.resource_group.name
+  login               = "sqladmin"
+  tenant_id           = data.azurerm_client_config.example_clientconfig.tenant_id
+  object_id           = data.azurerm_client_config.example_clientconfig.object_id
+}
+resource "azurerm_storage_account" "example" {
+  name                     = "accteststorageaccount"
+  resource_group_name      = azurerm_resource_group.example.name
+  location                 = azurerm_resource_group.example.location
+  account_tier             = "Standard"
+  account_replication_type = "GRS"
+}
+resource "azurerm_storage_container" "example" {
+  name                  = "accteststoragecontainer"
+  storage_account_name  = azurerm_storage_account.example.name
+  container_access_type = "private"
+}
+resource "azurerm_mssql_server_security_alert_policy" "example" {
```

Comment:
**Accurics remediation**
 
**[MEDIUM]: Ensure ‘connection_throttling' is enabled for Azure PostgreSQL Configuration**
 
**Vulnerability**
A feature, 'connection_throttling' is not used for Azure PostgreSQL Configuration this may lead to DoS attacks.
 
**Remediation**
Enabling connection_throttling helps the PostgreSQL Database to Set the verbosity of logged messages which in turn generates query and error logs with respect to concurrent connections, that could lead to a successful Denial of Service (DoS) attack by exhausting connection resources. A system can also fail or be degraded by an overload of legitimate users. Query and error logs can be used to identify, troubleshoot, and repair configuration errors and sub-optimal performance.

source: accurics-beta